src/lib/storage/segment_iterables.hpp

```diff
@@ -153,10 +153,17 @@ class PointAccessibleSegmentIterable : public SegmentIterable<Derived> {
     }
   }
 
+  // Needed, because otherwise the 2 overloads above are ambiguous
```

Comment:
two, also could you add half a sentence on why they are ambiguous?

Reply:
replaced with
"The two overloads of `with_iterators` are ambiguous if a `shared_ptr<PosListType>` is given.
This will call the overload expecting `shared_ptr<const PosListType>`"

Is that ok?

Reply:
Shouldn't the second version match `<PosListType = const WhateverPosList>`, too? Have you tried what happens if you only use the second version?

Reply:
Yes. I think we removed the initially ambiguous version of `with_iterators` that we wrote this resolver for. Using `shared_ptr<PosListType>` instead of `shared_ptr<const PosListType>` allows compiling without it.

Comment and resolve-method removed.